torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2431,54 +2431,90 @@ Tensor linalg_qr_backward(const std::vector<torch::autograd::Variable> &grads, c
   }
 }
 
-// Invertible case is derived from Jacobi's formula, and also can be found at:
-// http://eprints.maths.ox.ac.uk/1079/1/NA-08-01.pdf
 Tensor linalg_det_backward(const Tensor & grad, const Tensor& self, const Tensor& det) {
-  auto singular_case_backward = [&](const Tensor& grad, const Tensor& self, const Tensor& det) -> Tensor {
-    Tensor u, sigma, v;
-    std::tie(u, sigma, v) = self.svd();
-    auto gsigma = prod_backward(grad.unsqueeze(-1), sigma, det.unsqueeze(-1));
-    return svd_backward({{}, gsigma, {}}, self, true, true, u, sigma, v);
+  if (self.numel() == 0) {
+    return at::empty_like(self);
+  }
+  auto det_backward_nonsingular = [&](const Tensor& grad, const Tensor& self, const Tensor& det) -> Tensor {
+    // Derived from Jacobi's formula for partial derivative, which can be found
+    // at https://en.wikipedia.org/wiki/Jacobi%27s_formula
+
+    // Expand each scalar determinant into a diagonal matrix whose size matches
+    // the matrices in `self`. This is required for at::solve
+    auto det_sizes = det.sizes().vec();
+    auto det_strides = det.strides().vec();
+    det_sizes.push_back(self.size(-1));
+    det_strides.push_back(0);
+    auto det_vecs = det.as_strided(det_sizes, det_strides);
+    auto det_diag = det_vecs.diag_embed();
+    auto det_grad = std::get<0>(at::solve(det_diag, self.transpose(-1, -2))).conj();
```

Comment:
let probably use `linalg.solve`? The order of arguments is swapped and is more natural, actually.